app/controllers/api/sessions_controller.rb

```diff
@@ -0,0 +1,61 @@
+module Api
+  class SessionsController < ApplicationController
+    prepend_before_action :skip_session_expiration
+    prepend_before_action :skip_devise_hooks
+
+    respond_to :json
+
+    def show
+      render json: { active: active?, timeout: timeout }
+    end
+
+    def update
+      analytics.session_kept_alive if active?
+      update_last_request_at
+      render json: { active: active?, timeout: timeout }
+    end
+
+    def destroy
+      analytics.session_timed_out
```

Comment:
I'm observing some issues where this method is bypassed and the user is redirected to the root URL. I think it may be due to the CSRF token expiring, related to [a recent discussion in Slack](https://gsa-tts.slack.com/archives/C0NGESUN5/p1678740248343529?thread_ts=1678739597.630529&cid=C0NGESUN5) where extending the session doesn't regenerate/extend the CSRF on the page. This isn't a problem for the previous implementation because the `/timeout` route is handled as a `GET`, but this is a `DELETE`.

Reply:
This should be fixed now between #8067 and 63bbdc2 in this pull request.